src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -410,17 +457,27 @@ public SemanticVersion(string version)
         /// <param name="patch">The minor version</param>
         /// <param name="label">The label for the version</param>
         /// <exception cref="PSArgumentException">
-        /// If <paramref name="major"/>, <paramref name="minor"/>, or <paramref name="patch"/> is less than 0.
-        /// </exception>
-        /// <exception cref="PSArgumentNullException">
-        /// If <paramref name="label"/> is null or an empty string.
+        /// <exception cref="FormatException">
+        /// If <paramref name="label"/> starts with '-'.
```

Comment:
Complex RegEx is a puzzle for me too. I found some sample RegEx strings for SymVer but after testing I concluded that they all have bugs. 
So I split the SymVer regex string on some string (s_VersionSansRegEx, s_LabelRegEx, s_LabelUnitRegEx). This slightly increased the code but helped to avoid bugs.

Comment added.